airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,270 @@
+        submit = self._submit_dq_job(query)
+        check = self._check_dq_job_status(submit)
+        get = self._get_data_from_dq_job(check)
+        yield from get
+
+    # Warper for _get_data using date-slices as pages
+    def _get_data_with_date_slice(
+        self, q_type: str, obj: str, date_field: str, start_date: str, window_days: int, end_date: str = None
+    ) -> Iterable:
+        # Parsing input dates
+        s = pendulum.parse(start_date)
+        tz = self._get_tz(s)
+        # If there is no `end_date` as input - use now()
+        e = pendulum.parse(pendulum.now().to_datetime_string()) if end_date is None else pendulum.parse(end_date)
+        # check end_date bigger than start_date
+        if s >= e:
+            raise EndDateError
+        else:
+            # Get n of date-slices
+            n = self._get_n_slices(s, e, window_days)
+            # initiating slice_start/end for date slice
+            slice_start = s.to_datetime_string() + tz
+            # if slice_end is bigger than the input end_date, switch to the end_date
+            slice_end = self._check_end_date(self._next_slice_end_date(slice_start, window_days), e) + tz
+            # For multiple date-slices
+            if n > 1:
+                while n > 0:
+                    yield from self._get_data(q_type, obj, date_field, slice_start, slice_end)
+                    # make next date-slice
+                    slice_start = slice_end
+                    # if next date-slice end_date is bigger than the input end_date, we switch to the end_date
+                    slice_end = self._check_end_date(self._next_slice_end_date(slice_end, window_days), e) + tz
+                    # Modify iterator
+                    n -= 1
+            else:
+                # For 1 date-slice
+                yield from self._get_data(q_type, obj, date_field, slice_start, slice_end)
+
+    # Check the end-date before assign next date-slice
+    @staticmethod
+    def _check_end_date(slice_end_date: str, global_end_date: str) -> str:
+        return global_end_date.to_datetime_string() if pendulum.parse(slice_end_date) > global_end_date else slice_end_date
+
+    # Get n of date-slices needed to fetch the data
+    @staticmethod
+    def _get_n_slices(start_date: datetime, end_date: datetime, window_days: int) -> int:
+        # case where we have 1 sec difference between start and end dates would produce 0 slices, so return 1 instead
+        d = pendulum.period(start_date, end_date).in_days()
+        return 1 if ceil(d / window_days) <= 0 else ceil(d / window_days)
+
+    # Creates the next-date-slice
+    @staticmethod
+    def _next_slice_end_date(end_date: str, window_days: int) -> str:
+        format = "YYYY-MM-DD HH:mm:ss Z"
+        return pendulum.from_format(end_date, format).add(days=window_days).to_datetime_string()
+
+    # Get/Set the timezone information from the date_field
+    @staticmethod
+    def _get_tz(start_date: datetime) -> str:
```

Comment:
it's needed because we need to sync the information using TZ information that we received from the stream, otherwise, we will be using UTC and once we hit the incremental refresh we would get an incorrect number of records, due to timezone shift. Tested. 
I don't see much potential for the client doing so in the initial config of the connector, since this information could be set up wrong. I prefer to left it as is, it's giving the possibility to determine the TZ information right from the output data.